framework_tvb/tvb/core/adapters/abcuploader.py

```diff
@@ -77,8 +94,68 @@ def _prelaunch(self, operation, uid=None, available_disk_space=0, view_model=Non
         self.meta_data.update({DataTypeMetaData.KEY_SUBJECT: view_model.data_subject})
         self.generic_attributes.subject = view_model.data_subject
 
+        trait_upload_field_names = list(self.get_form_class().get_upload_information().keys())
+        for upload_field_name in trait_upload_field_names:
+            self._decrypt_content(view_model, upload_field_name)
+
         return ABCSynchronous._prelaunch(self, operation, uid, available_disk_space, view_model, **kwargs)
 
+    def _decrypt_content(self, view_model, trait_upload_field_name):
+        if view_model.encrypted_aes_key is None:
+            return
+
+        if TvbProfile.current.UPLOAD_KEY_PATH is None or not os.path.exists(TvbProfile.current.UPLOAD_KEY_PATH):
+            raise LaunchException("We can not process Encrypted files at this moment, "
+                                  "due to missing PK for decryption! Please contact the administrator!")
+
+        # Open encrypted file
+        upload_path = getattr(view_model, trait_upload_field_name)
+        with open(upload_path, 'rb') as f:
+            encrypted_content = f.read()
```

Comment:
Will this full read of the uploaded file be safe in case of very large files (e.g. surfaces) ?
Could we buffer the encryption somehow ?